From ccedff2f357d7d52bf4327cf14dfade2b41052ca Mon Sep 17 00:00:00 2001
From: Kjetil Matheussen <k.s.matheussen@notam02.no>
Date: Sat, 15 Jul 2017 13:21:17 +0200
Subject: [PATCH] JackDebugClient: Keep client->fSynchroTable. Prevents crash
 in jack_client_wait, plus that it seems to be the correct thing to do.

---
 common/JackClient.cpp      | 3 ---
 common/JackClient.h        | 1 -
 common/JackDebugClient.cpp | 1 +
 3 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/common/JackClient.cpp b/common/JackClient.cpp
index 7334a57a..bcb06ead 100644
--- a/common/JackClient.cpp
+++ b/common/JackClient.cpp
@@ -39,9 +39,6 @@ namespace Jack
 
 #define IsRealTime() ((fProcess != NULL) | (fThreadFun != NULL) | (fSync != NULL) | (fTimebase != NULL))
 
-JackClient::JackClient():fThread(this)
-{}
-
 JackClient::JackClient(JackSynchro* table):fThread(this)
 {
     fSynchroTable = table;
diff --git a/common/JackClient.h b/common/JackClient.h
index 3013334b..b6e78ef3 100644
--- a/common/JackClient.h
+++ b/common/JackClient.h
@@ -124,7 +124,6 @@ class SERVER_EXPORT JackClient : public JackClientInterface, public JackRunnable
 
     public:
 
-        JackClient();
         JackClient(JackSynchro* table);
         virtual ~JackClient();
 
diff --git a/common/JackDebugClient.cpp b/common/JackDebugClient.cpp
index 3308c457..7c1f539a 100644
--- a/common/JackDebugClient.cpp
+++ b/common/JackDebugClient.cpp
@@ -35,6 +35,7 @@ namespace Jack
 {
 
 JackDebugClient::JackDebugClient(JackClient * client)
+  : JackClient(client->fSynchroTable)
 {
     fTotalPortNumber = 1;       // The total number of port opened and maybe closed. Historical view.
     fOpenPortNumber = 0;        // The current number of opened port.
-- 
2.15.0

